ResearchKit/Common/ORKPasscodeStepViewController.m

```diff
@@ -647,4 +672,26 @@ - (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
     return _shouldResignFirstResponder;
 }
 
+- (void)forgotPasscodeTapped
+{
+    if ([self.passcodeDelegate respondsToSelector:@selector(passcodeViewControllerForgotPasscode:)])
+    {
+        [self.passcodeDelegate passcodeViewControllerForgotPasscode:self];
+    }
+}
+
+#pragma mark - Keyboard Notifications
+
+- (void)keyboardWillShow:(NSNotification *)notifcation
+{
+    CGFloat keyboardHeight = [notifcation.userInfo[UIKeyboardFrameEndUserInfoKey] CGRectValue].size.height;
+    
+    double animationDuration = [notifcation.userInfo[UIKeyboardAnimationDurationUserInfoKey] doubleValue];
+    
+    [UIView animateWithDuration:animationDuration animations:^
+    {
+        [_forgotPasscodeButton setFrame:CGRectMake(_forgotPasscodeButton.frame.origin.x, _originalForgotPasscodeY - keyboardHeight, _forgotPasscodeButton.frame.size.width, _forgotPasscodeButton.frame.size.height)];
+    }];
+}
+
```

Comment:
Currently, it can't.  However, I will put in a comment tomorrow that says, if UI is added for keyboard going up, "Forgot Passcode" button will need to animate back to the bottom.